Sources/StreamChat/Database/DTOs/MemberModelDTO.swift

```diff
@@ -91,17 +110,60 @@ extension MemberDTO {
 }
 
 extension NSManagedObjectContext {
-    func saveMember(
-        payload: MemberPayload,
-        channelId: ChannelId,
-        query: ChannelMemberListQuery?
-    ) throws -> MemberDTO {
-        let dto = MemberDTO.loadOrCreate(id: payload.user.id, channelId: channelId, context: self)
+    func upsertMany(payload: [MemberPayload], channelId: ChannelId) throws -> [MemberDTO] {
+        // the IDs that we want to have sorted as comes from payload
+        let userIDs = payload.map(\.user.id)
+
+        var payloadById = [String: MemberPayload]()
+        payload.forEach {
+            payloadById[$0.user.id] = $0
+        }
+
+        var membersByUserID = [String: MemberDTO]()
+
+        // fetch all members based on their IDs
+        let existingMembers = try fetch(MemberDTO.members(userIDs, in: channelId))
+        existingMembers.forEach {
+            // tell core data that we do not want to merge any changes for these objects (sigh)
```

Comment:
Not clear to me why this is done in every upsert. Could you expand?

Reply:
if the object is not refreshed, core data will write it as it is after the read. the same thing happens if I read a subset of fields. Alternative approach would be to populate all objects read from core data, but that's a bit annoying and expensive